Port to Nuvola Player 3

thenyeguy opened the pull request:
I used the old Spotify plugin to get the scraping and rewrote the plugin for Nuvola Player 3. It only handles the core functionality - buttons/media keys and metadata.

jiri-janousek commented:
Hi @thenyeguy. Thanks for your work. I will probably take a look at it in February after I release Nuvola Player 3 beta.

jiri-janousek commented on Makefile:
```diff
@@ -22,15 +22,33 @@
 # Default installation destination
 DEST ?= $(HOME)/.local/share/nuvolaplayer3/web_apps
+# Size of PNG app icon
+ICON_SIZE ?= 48
```
Change [icon size in Makefile to 128](https://groups.google.com/d/msg/nuvola-player-devel/a-axnC05sf8/a8hEsBolxiIJ).

jiri-janousek commented on README.md:
```diff
@@ -3,7 +3,9 @@ Spotify for Nuvola Player 3
 [Nuvola Player](https://github.com/tiliado/nuvolaplayer).
- 
+
+Based on the Nuvola Player 2 Spotify plugin by Chase Colman (<chase@colman.io>).
```
How much is your work based on Chase's original script? If there are significant code portions from his script, you have to retain copyright from the original file.

thenyeguy replied:
Good point. I only used his script as a reference to see how to get information from Spotify's API. I'm not familiar with licensing stuff at all, should the original copyright be retained?

jiri-janousek replied:
It's ok if you used his script only as a reference.

jiri-janousek commented on integrate.js:
```diff
@@ -0,0 +1,179 @@
+        // Connect handler for signal ActionActivated
+        Nuvola.actions.connect("ActionActivated", this);
+
+        // this.connect();
```
Remove commented-out code.

jiri-janousek commented on metadata.json:
```diff
@@ -1,9 +1,10 @@
+    "maintainer_name": "Michael Nye",
+    "maintainer_link": "https://github.com/thenyeguy",
+    "home_url": "https://play.spotify.com/browse",
+    "version_major": 1,
```
Revert version back to 2.0. 1.0 has been already released.

jiri-janousek commented:
I've added some in-line comments. I haven't tested whether your integration actually works because I cannot make Spotify work behind HTTP proxy.

Also, could you fill in white inner parts of the Spotify logo? I think it would look better ;-)

thenyeguy commented:
I fixed the issues you mentioned, thanks for the feedback. The log also does look better with white!

However, I updated Nuvola Player for the first time in a while, and Spotify no longer loads on my machine. It seems to happen even with the template integrate.js on the development tutorial. I made a [gist with the debug output](https://gist.github.com/thenyeguy/e0dc7aa126a571bef427). There is one javascript error, but I don't see the same error in the console when loading Spotify in Chrome or Firefox.

It seems like a Nuvola issue, but I can't tell for sure. I could open this as an issue against Nuvola?

jiri-janousek commented:
> However, I updated Nuvola Player for the first time in a while, and Spotify no longer loads on my machine. It seems to happen even with the template integrate.js on the development tutorial. I made a gist with the debug output. There is one javascript error, but I don't see the same error in the console when loading Spotify in Chrome or Firefox.
> 
> It seems like a Nuvola issue, but I can't tell for sure. I could open this as an issue against Nuvola?

It works fine on my machine. [Create a new bug report](https://github.com/tiliado/nuvola-app-spotify/issues/new) for that. And also check whether your issue is reproducible in WebKit2 MiniBrowser. If you use Debian/Ubuntu, it should be available in one of these locations:

```
/usr/lib/x86_64-linux-gnu/webkit2gtk-3.0/libexec/MiniBrowser
/usr/lib/x86_64-linux-gnu/webkit2gtk-4.0/libexec/MiniBrowser
/usr/lib/i386-linux-gnu/webkit2gtk-3.0/libexec/MiniBrowser
/usr/lib/i386-linux-gnu/webkit2gtk-4.0/libexec/MiniBrowser
```

I will merge your branch today/tomorrow and grant you commit access to [the main repository](https://github.com/tiliado/nuvola-app-spotify). If you haven't already done so:
1. Subscribe to the [Nuvola Player Development forum](https://groups.google.com/d/forum/nuvola-player-devel) as I post here announcements related to development.
2. Create an account at [Tiliado.eu](https://tiliado.eu) and send me your username. I'll add you to Tiliado developers group. You can then fill in your credentials in Nuvola Player preferences not to be bothered by donate bar.
